Dockerfile

```diff
@@ -20,15 +26,35 @@ RUN npm ci --no-audit
 # Set commit hash
 RUN if [[ $IS_STABLE == "0" ]] ; then NUXT_ENV_COMMIT=$(git rev-parse HEAD) ; fi
 
+
 # Build client
 RUN npm run build
 
+
 # Deploy built distribution to nginx
 FROM nginx:alpine
+
+# Set environment variables
+ARG HISTORY_ROUTER_MODE=false
+
 COPY --from=build /app/src/dist/ /usr/share/nginx/html/
+
+COPY --from=build /app/.docker/nginx.conf /etc/nginx/conf.d/custom.conf 
+
+# If specified, set's custom nginx.conf to support spa (for 'history' routing mode)
```

Comment:
I would honestly remove all this block and just copy our custom config no matter the history mode. The default config gives the same end result to the user (the only difference is the try block) and this just clutters our Dockerfile for the little benefit it provides 

*(now at least, maybe in the future it's useful for E2E testing)*

Thoughts @ThibaultNocchi?

Reply:
I was thinking that a custom nginx.conf whoud be a problem for some cases [(your comment)](https://github.com/jellyfin/jellyfin-vue/pull/1685#issuecomment-1009509281)

No?

Reply:
Oh yes, but I meant users that dont use Docker (not everyone does).  
If an user builds the client himself or pulls it as a deb (like current Web) or from a release (we don't have debs or releases, but once we reach an stable state we will) and serves the client through his webserver of choice, the user must configure it with the ``try_files`` directive.


Docker in that regard its our walled garden and we can provide our best config where everything is beautiful. But we can't expect everybody will have our ideal configuration, hence the need for the default to be hash. 

In this repo we are "selling" users two "bundles":

* The client (our "best-seller", as its what the repo is about) 

* Client + webserver (Docker) 

This repo is all about the client and, as is, it should work as universally as possible with the default config. History mode doesn't allow that as it adds the user's webserver into the mix, and adds variables out of our control. An user with a misconfigured webserver will have a degraded experience, so we should stick to "the universal best". 

However, we're offering too (through Docker) the mix client + webserver: with it, we remove the webserver variable from the user's hands and bring it to us. This means that for users choosing our bundle, we can give them what we think are the best defaults (we all agree is history mode). 

To sum up: the file does not hurt and provides "universally" safe defaults. I also expect nobody in Docker will use the hash mode, because it makes no sense for them (most people just fire docker build or pull the image, and those will use history mode by default). 
I only expect people building outside Docker to care about changing that option, since those are the ones who are likely not going to use "our walled garden" and are more likely to be scatterbrained about their webserver config. 

That's why the current config is ideal: outside docker we use hash, inside it we use history. The environments where they both make more sense.

Reply:
Then in this case it is better to change the default ARG to true, 
to pass the --build-arg only in the unlikely event that you want to disable. Do you agree?

bc now default is 'hash' for docker too

Reply:
I agree that we can copy the config in any case, it's basic SPA routing anyway. :)